cobald_tests/daemon/core/test_config.py

```diff
@@ -13,6 +15,60 @@
 COBalDLoader.add_constructor(tag="!MockPool", constructor=yaml_constructor(MockPool))
 
 
+class TagTracker:
+    """Helper to track the invocation of YAML !Tags"""
+
+    count = -1
+
+    def __init__(self, *args, **kwargs):
+        type(self).count = self.count = type(self).count + 1
+        self.orig_args = copy.deepcopy(args)
+        self.orig_kwargs = copy.deepcopy(kwargs)
+        self.args = args
+        self.kwargs = kwargs
+
+    @classmethod
+    @contextmanager
+    def scope(cls, base_count=-1):
+        old_count, cls.count = cls.count, base_count
+        try:
+            yield
+        finally:
+            cls.count = old_count
+
+
+COBalDLoader.add_constructor(
+    tag="!TagTracker", constructor=yaml_constructor(TagTracker, eager=True)
+)
+COBalDLoader.add_constructor(
+    tag="!TagTrackerLazy", constructor=yaml_constructor(TagTracker, eager=False)
+)
+
+
+@yaml_constructor(eager=True)
+def full_deco_constructor(*args, **kwargs):
+    return copy.deepcopy(args), copy.deepcopy(kwargs)
+
+
+@yaml_constructor
+def bare_deco_constructor(*args, **kwargs):
+    return copy.deepcopy(args), copy.deepcopy(kwargs)
```

Comment:
How does the naming tell me what is done? The only difference I can spot is that one is used lazy and the other eager. Why is one then called `full` and the other `bare`?